### PR TITLE
Add tests for check_branches.sh (the two scripts that run in the docker for checking branches).

### PR DESCRIPTION
**Pull request by fermadeiral**
This PR proposes tests for the scripts that checks branches, the ones that run in the docker: `bears-checkbranches-dockerimage/check_branches.sh` (Bears) and `checkbranches-dockerimage/check_branches.sh` (Repairnator). For each script, 3 tests were added:

1) on a valid branch, so the result of the test should be [OK].
2) on a branch without the json file, so the result of the test should be [FAILURE] (<name of the file>.json does not exist)
3) on a branch with an invalid json file, so the result of the test should be [FAILURE] (<name of the file>.json is invalid)

For Repairnator, the first test that should be OK is failing: the issue is that the branch I generated yesterday by running repairnator contains a `repairnator.json` file that does not match the `repairnator-schema.json`, so this is a bug when generating the `repairnator.json`, which is a different issue that should be handle in another PR.

Additionally, it seems possible to run the tests that this PR adds in Travis: https://www.sysorchestra.com/2014/10/12/travisshunit2/ . I still didn't try it, because that would involve changes in the `.travis.yml` file, and I don't know if you would agree with that, so let me know if you would like to run these tests on Travis.

### CODE REVIEW

**Comment by surli** [comment]
Those tests implies that shunit2 is available in the current directory. So at least add some documentation to setup this properly.

**Comment by fermadeiral** [comment]
> Those tests implies that shunit2 is available in the current directory.

Not really in the current dir, but in the machine.

> So at least add some documentation to setup this properly.

I don't follow you. You want I add a documentation on how to run the tests locally by hands?

And also what about Travis? You said nothing about it, so I don't really know...

**Comment by surli** [comment]
> I don't follow you. You want I add a documentation on how to run the tests locally by hands?

Those tests are not standard junit tests, so I don't really know how to execute them, what I should install, etc.
So yeah just add a line of comment to point out what I need to install/how and to tell me how to run them.

> And also what about Travis?

I'm ok to add it on Travis if you want yes.

**Comment by fermadeiral** [comment]
>  So yeah just add a line of comment to point out what I need to install/how and to tell me how to run them.

Simon, I added a README in the two folders that contain this type of test. The steps I described there are the ones I used to run the tests. I believe you can reproduce the tests now.

> I'm ok to add it on Travis if you want yes.

Ok, I will try it, I'm not sure if this will work. Just for the record, if this works, the build will fail because of the issue #701, so we will need to put this PR on hold until we fix the issue.

File: repairnator/docker-images/bears-checkbranches-dockerimage/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Testing `check_branches.sh`
+
+The tests on `check_branches.sh` are in the file `check_branches_test.sh`, which relies on the testing framework [shUnit2](https://github.com/kward/shunit2).
+To run the tests, two steps are required:
+
+1. Install shUnit2:
+
+```bash
+$ sudo apt-get install shunit2
```
Comment (surli) [inline repairnator/docker-images/bears-checkbranches-dockerimage/README.md]: Could you modify with:

```
$ sudo apt-get install shunit2 # for debian/ubuntu linux
$ brew install shunit2 # for macos
```

Comment (fermadeiral) [reply repairnator/docker-images/bears-checkbranches-dockerimage/README.md]: Sure. I will do that after I finish my fight with the Travis build.

**Comment by fermadeiral** [comment]
The work here is done. It's running on Travis.

The build is broken because of issue #701.
In the last case, I propose we remove (or comment the call of) the execution of the tests in `checkbranches-dockerimage/check_branches_test.sh` on Travis, because the work is not done yet in the checkbranches module for Bears, and I will need the tests for Bears added in this PR. So please, let me know your plans for issue #701 so I can plan what I do.

**Comment by fermadeiral** [comment]
I will create new branches for testing the check branches for Repairnator so I can make the build passing here.

**Comment by surli** [comment]
Not sure to understand: why don't you merge the master in this branch?

**Comment by fermadeiral** [comment]
> Not sure to understand: why don't you merge the master in this branch?

My bad, I did a confusion here in my mind. What I need to do is to merge the master here and to update the schema in the docker-images folder.

**Comment by fermadeiral** [comment]
@surli it's finally ready for merging.

**Pull request merged**